src/common/types/auth.ts

```diff
@@ -62,13 +62,16 @@ export type Group = {
 	name: string;
 };
 
-export type UserDataResponseErrorType = {
-	type: string;
-	title: string;
-	detail: string;
-	status: number;
-	error: string;
-};
+export type UserDataResponseErrorType =
+	| {
+			type: string;
+			title: string;
+			detail: string;
+			status: number;
+	  }
+	| {
+			error: string;
+	  };
```

Comment:
Just FYI there's no discriminator here. The consuming funciton will need to do a check like `if('error' in responseError) {}`.